src/sensor/parser.h

```diff
@@ -0,0 +1,25 @@
+#pragma once
+
+#include <QObject>
+#include "ping-message.h"
+
+// This class digests data and notifies owner when something interesting happens
+class Parser : public QObject
+{
+    Q_OBJECT
+public:
+    Parser() = default;
+    ~Parser() = default;
+
+    uint32_t parsed; // number of messages/packets successfully parsed
+    uint32_t errors; // number of parse errors
+
+    // asynchronous use, Child classes should signal when something happens ie. 'emit newMessage(Message m)'
+    virtual void parseBuffer(const QByteArray& data) { Q_UNUSED(data) }
+    // synchronous use, Child should return flags indicating incremental parse result/status
+    virtual uint8_t parseByte(const char byte) { Q_UNUSED(byte) return 0; }
+
```

Comment:
The class should return somehow a synchronous way to get the last parsed message.

Reply:
done

Reply:
Why `parseBuffer` does not return state ? This will obligate us to create a loop and do what `parBuffer` is already doing.